libs/coin-modules/coin-cardano/src/js-signOperation.ts

```diff
@@ -206,133 +191,59 @@ const buildOptimisticOperation = (
-            if (rcert.certType === (CertificateType.STAKE_REGISTRATION as number)) {
-              return prepareStakeRegistrationCertificate(
-                rcert as TyphonTypes.StakeRegistrationCertificate,
-              );
-            } else if (rcert.certType === (CertificateType.STAKE_DELEGATION as number)) {
-              return prepareStakeDelegationCertificate(
-                rcert as TyphonTypes.StakeDelegationCertificate,
-              );
-            } else if (rcert.certType === (CertificateType.STAKE_DEREGISTRATION as number)) {
-              return prepareStakeDeRegistrationCertificate(
-                rcert as TyphonTypes.StakeDeRegistrationCertificate,
-              );
-            } else {
-              throw new Error("Invalid Certificate type");
-            }
-          });
-
-          const rawWithdrawals = unsignedTransaction.getWithdrawals();
-          const ledgerWithdrawals = rawWithdrawals.map(prepareWithdrawal);
-
-          const auxiliaryDataHashHex = unsignedTransaction.getAuxiliaryDataHashHex();
-
-          const networkParams = getNetworkParameters(account.currency.id);
-          const network =
-            networkParams.networkId === Networks.Mainnet.networkId
-              ? Networks.Mainnet
-              : Networks.Testnet;
-
-          const trxOptions: SignTransactionRequest = {
-            signingMode: TransactionSigningMode.ORDINARY_TRANSACTION,
-            tx: {
-              network,
-              inputs: ledgerAppInputs,
-              outputs: ledgerAppOutputs,
-              certificates: ledgerCertificates,
-              withdrawals: ledgerWithdrawals,
-              fee: unsignedTransaction.getFee().toString(),
-              ttl: unsignedTransaction.getTTL()?.toString(),
-              validityIntervalStart: null,
-              auxiliaryData: auxiliaryDataHashHex
-                ? {
-                    type: TxAuxiliaryDataType.ARBITRARY_HASH,
-                    params: {
-                      hashHex: auxiliaryDataHashHex,
-                    },
-                  }
-                : null,
-            },
-            additionalWitnessPaths: [],
-          };
+const buildSignOperation =
+  (
+    signerContext: SignerContext<
+      CardanoSigner,
+      CardanoAddress | CardanoExtendedPublicKey | CardanoSignature
+    >,
+  ): SignOperationFnSignature<Transaction> =>
+  ({ account, deviceId, transaction }): Observable<SignOperationEvent> =>
+    new Observable(o => {
+      async function main() {
+        o.next({ type: "device-signature-requested" });
+
+        if (!transaction.fees) {
+          throw new FeeNotLoaded();
+        }
 
-          // Sign by device
-          const appAda = new Ada(transport);
-          const r = await appAda.signTransaction(trxOptions);
-          const signed = signTx(unsignedTransaction, accountPubKey, r.witnesses);
+        const unsignedTransaction = await buildTransaction(account as CardanoAccount, transaction);
```

Comment:
No way to actually make sure that the account passed as a parameter is of type `CardanoAccount` ?

Reply:
To fix this, it will imply to touch all CoinModule and `@legerhq/types-live`, which is beyond this PR. But good point of improvement to add into the team's backlog.